dist/tools/ng_sniffer/ng_sniffer.py

```diff
@@ -90,38 +44,100 @@
-
-        pkt_header = re.match(r">? *rftest-rx --- len 0x(\w\w).*", line)
+NETWORK = 230       # 802.15.4 no FCS
+
+
+def configure_interface(port, channel):
+    line = ""
+    iface = 0
+    port.write(b'ifconfig\n')
+    while True:
+        line = port.readline()
+        if line == '':
+            print("Application has no network interface defined",
+                  file=sys.stderr)
+            sys.exit(2)
+        match = re.search(r'^Iface +(\d+)', line.decode())
+        if match is not None:
+            iface = int(match.group(1))
+            break
+
+    # set channel, raw mode, and promiscuous mode
+    print('ifconfig %d set chan %d' % (iface, channel), file=sys.stderr)
+    print('ifconfig %d raw' % iface, file=sys.stderr)
+    print('ifconfig %d promisc' % iface, file=sys.stderr)
+    port.write(('ifconfig %d set chan %d\n' % (iface, channel)).encode())
+    port.write(('ifconfig %d raw\n' % iface).encode())
+    port.write(('ifconfig %d promisc\n' % iface).encode())
+
+
+def generate_pcap(port, out):
+    # count incoming packets
+    count = 0
+    # output overall PCAP header
+    out.write(pack('<LHHLLLL', MAGIC, MAJOR, MINOR, ZONE, SIG, SNAPLEN,
+                   NETWORK))
+    sys.stderr.write("RX: %i\r" % count)
+    while True:
+        line = port.readline().rstrip()
+
+        pkt_header = re.match(r">? *rftest-rx --- len 0x(\w\w).*",
+                              line.decode())
         if pkt_header:
-            t = time.time()
-            sec = int(t)
-            usec = (t - sec) * 100000
+            now = time()
+            sec = int(now)
+            usec = (int(now) - sec) * 100000
             length = int(pkt_header.group(1), 16)
-            outfile.write(pack('<LLLL', sec, usec, length, length))
-            outfile.flush()
+            out.write(pack('<LLLL', sec, usec, length, length))
+            out.flush()
             count += 1
             sys.stderr.write("RX: %i\r" % count)
             continue
 
-        pkt_data = re.match(r"(0x\w\w )+", line)
+        pkt_data = re.match(r"(0x\w\w )+", line.decode())
         if pkt_data:
-            for part in line.split(' '):
+            for part in line.decode().split(' '):
                 byte = re.match(r"0x(\w\w)", part)
                 if byte:
-                    outfile.write(pack('<B', int(byte.group(1), 16)))
-            outfile.flush()
+                    out.write(pack('<B', int(byte.group(1), 16)))
+            out.flush()
+
+
+def main(argv):
+    if len(argv) < 4:
+        print("Usage: %s tty baudrate channel [outfile]" % (argv[0]),
+              file=sys.stderr)
+        print("       channel = 11-26", file=sys.stderr)
+        sys.exit(2)
+
+    # open serial port
+    try:
+        serport = Serial(argv[1], argv[2], dsrdtr=0, rtscts=0,
+                         timeout=1)
+    except IOError:
+        print("error opening port", file=sys.stderr)
+        sys.exit(2)
+
+    sleep(1)
```

Comment:
remove the sleep if not needed, otherwise add a comment in the code why it is necessary to sleep.